### ISSUE TITLE
Improve Search responsiveness

### ISSUE THREAD

**Issue by di72nn** [wallabag/android-app#490]
There are two obvious options:
 * Use "submit" button for Search field instead of performing search on every text change.
 * If the text in the search field changes, don't immediately update list, but schedule an update with a short delay. If the text is changed again (shortly after that), reschedule the update.

The options can be combined.

Current implementation contains hacks that ~~preserve the search field state~~ open new search field with the same text when context menu is recreated, so it should be done carefully.

**Comment by Suncatcher** [comment]
Support this, and I am for the first option.
Regarding second option: the idea is quite sensible if all computing is done *in the background*, so to not freeze the searchbox while querying the DB. Though I don't know if it's possible in Symfony.

The other option is to set *type delay*, i.e. when user is typing the word (=continuously) do not do the search, however when he typed the word (N seconds passed) start the search.

**Comment by di72nn** [comment]
> Though I don't know if it's possible in Symfony.

We're talking about Android here, so the server backend doesn't matter.

> The other option is to set type delay, i.e. when user is typing the word (=continuously) do not do the search, however when he typed the word to end (N seconds passed), then start the search.

Pretty much what I tried to describe by my second option. Thanks for a clearer version.

**Comment by Suncatcher** [comment]
> We're talking about Android here, so the server backend doesn't matter.

So the bottleneck here is Android DB-stack? Wallabag app store the articles in Android SQLite?

**Comment by di72nn** [comment]
> So the bottleneck here is Android DB-stack? Wallabag app store the articles in Android SQLite?

Yes, the articles are stored in an SQLite DB. One of the problems with search responsiveness (and any lists updates responsiveness, in general) is the fact that the lists are updated on the UI (main) thread (which causes UI freezes). So the DB performance is not the exactly the problem.

**Comment by Suncatcher** [comment]
The real place where incremental search should really be (and where we lack it) is a web-version search :)
I suppose computational power of any server allows implementing this.

**Comment by tenkabuto** [comment]
I would like to suggest that the OP's first option should be implemented. Automatically searching is quite resource-intensive on my phone.
